src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    class Node<T> {
+
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node node = new Node(null, value, null);
+        if (head == null) {
+            head = node;
+            tail = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexExist(index);
+        Node<T> currentNode = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                if (currentNode == null) {
+                    add(value);
+                } else {
+                    if (currentNode.prev == null) {
+                        Node<T> newNode = new Node<>(null, value, head);
+                        head.prev = newNode;
+                        head = newNode;
+                    } else {
+                        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+                        currentNode.prev.next = newNode;
+                        currentNode.prev = newNode;
+                    }
+                    size++;
+                }
+                break;
+            }
+            currentNode = currentNode.next;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+            currentNode = currentNode.next;
+        }
```

Comment:
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```

why do we need this?